src/sqlfluff/dialects/dialect_redshift.py

```diff
@@ -56,3 +65,214 @@ class FunctionSegment(BaseSegment):
+        ),
+    )
+
+
+@redshift_dialect.segment()
+class TableAttributeSegment(BaseSegment):
+    """Redshift specific table attributes.
+
+    As specified in https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_TABLE_NEW.html
+    """
+
+    type = "table_constraint_segment"
+
+    match_grammar = AnyNumberOf(
+        Sequence("DISTSTYLE", OneOf("AUTO", "EVEN", "KEY", "ALL"), optional=True),
+        Sequence("DISTKEY", Bracketed(Ref("ColumnReferenceSegment")), optional=True),
+        OneOf(
+            Sequence(
+                OneOf("COMPOUND", "INTERLEAVED", optional=True),
+                "SORTKEY",
+                Bracketed(Delimited(Ref("ColumnReferenceSegment"))),
+            ),
+            Sequence("SORTKEY", "AUTO"),
+            optional=True,
+        ),
+        Sequence("ENCODE", "AUTO", optional=True),
+    )
+
+
+@redshift_dialect.segment(replace=True)
+class TableConstraintSegment(BaseSegment):
+    """Redshift specific table constraints.
+
+    As specified in https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_TABLE_NEW.html
+    """
+
+    type = "table_constraint_segment"
+
+    match_grammar = AnyNumberOf(
+        Sequence(
+            "UNIQUE", Bracketed(AnyNumberOf(Delimited(Ref("ColumnReferenceSegment"))))
+        ),
+        Sequence(
+            "PRIMARY",
+            "KEY",
+            Bracketed(AnyNumberOf(Delimited(Ref("ColumnReferenceSegment")))),
+        ),
+        Sequence(
+            "FOREIGN",
+            "KEY",
+            Bracketed(AnyNumberOf(Delimited(Ref("ColumnReferenceSegment")))),
+            "REFERENCES",
+            Ref("TableReferenceSegment"),
+            Sequence(Bracketed(Ref("ColumnReferenceSegment"))),
+        ),
+    )
+
+
+@redshift_dialect.segment(replace=True)
+class LikeOptionSegment(BaseSegment):
+    """Like Option Segment.
+
+    As specified in https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_TABLE_NEW.html
+    """
+
+    type = "like_option_segment"
+
+    match_grammar = Sequence(OneOf("INCLUDING", "EXCLUDING"), "DEFAULTS")
+
+
+@redshift_dialect.segment(replace=True)
+class CreateTableStatementSegment(BaseSegment):
+    """A `CREATE TABLE` statement.
+
+    As specified in https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_TABLE_NEW.html
+    """
+
+    type = "create_table_statement"
+
+    match_grammar = Sequence(
+        "CREATE",
+        Sequence(
+            Sequence("LOCAL", optional=True),
+            Ref("TemporaryGrammar", optional=True),
+            optional=True,
+        ),
+        "TABLE",
+        Ref("IfNotExistsGrammar", optional=True),
+        Ref("TableReferenceSegment"),
+        Bracketed(
+            OneOf(
+                # Columns and comment syntax:
+                Delimited(
+                    Sequence(
+                        Ref("ColumnReferenceSegment"),
+                        Ref("DatatypeSegment"),
+                        AnyNumberOf(Ref("ColumnAttributeSegment", optional=True)),
```

Comment:
Can you move the optional=True for this and the line below into the AnyNumberOf() rather than the Ref()